### PR TITLE
Add & like buttons on item page

### PR DESCRIPTION
**Pull request by alansutherland**

### CODE REVIEW

File: components/item/ItemHero.vue [hunk 1]
```diff
@@ -21,7 +21,13 @@
           >
             {{ rightsStatement }}
           </span>
-          <div>
+          <div class="d-flex">
+            <UserButtons
+              v-if="showUserButtons"
+              v-model="identifier"
+              @like="$emit('like', identifier)"
```
Comment (rwd) [inline components/item/ItemHero.vue:28]: These two event emissions are redundant, I think. Remove them if you concur.

**Review by rwd — CHANGES_REQUESTED** [review]

File: components/item/ItemHero.vue (hunk 1 above)
Comment (alansutherland) [reply components/item/ItemHero.vue]: I agree, removed

**Review by rwd — APPROVED** [review]

File: components/item/ItemHero.vue [hunk 2]
```diff
@@ -95,6 +100,9 @@
       },
       downloadEnabled() {
         return this.rightsStatement && !this.rightsStatement.includes('/InC/');
+      },
+      showUserButtons() {
+        return Boolean(Number(process.env.ENABLE_XX_USER_AUTH)) && Boolean(Number(process.env.ENABLE_UNAUTHENTICATED_USER_BUTTONS));
```
Comment (lbiedinger) [inline components/item/ItemHero.vue:105]: This won't show the button for logged in users now, add:
`|| (this.$store.state.auth && this.$store.state.auth.loggedIn));`
To bring it in line with this https://github.com/europeana/portal.js/blob/master/components/item/ItemPreviewCard.vue#L78

File: components/account/UserButtons.vue [hunk 3]
```diff
@@ -82,7 +82,11 @@
         this.$bvModal.show(this.addItemToSetModalId);
       },
       async toggleLiked() {
-        await (this.liked ? this.unlike() : this.like());
+        if (this.$auth.loggedIn) {
+          await (this.liked ? this.unlike() : this.like());
+        } else {
+          this.$auth.loginWith('keycloak');
+        }
```
Comment (lbiedinger) [inline components/account/UserButtons.vue:89]: I don't think this change is needed.
Attempting to like an item while not logged in will trigger the authentication in the store.
Revert this and the changes to the specs, unless there's a case I'm missing?

**Review by lbiedinger — CHANGES_REQUESTED** [review]

File: components/item/ItemHero.vue (hunk 2 above)
Comment (alansutherland) [reply components/item/ItemHero.vue:105]: oops, ok added

File: components/account/UserButtons.vue (hunk 3 above)
Comment (alansutherland) [reply components/account/UserButtons.vue:89]: it was showing me an error when I was logged out, but now I'm unable to replicate. Reverting

**Review by lbiedinger — APPROVED** [review]

**Pull request merged**